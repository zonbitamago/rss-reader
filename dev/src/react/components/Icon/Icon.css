.Icon {
  color: white;
  width: inherit;
  height: inherit;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Icon > svg {
  width: inherit;
  height: inherit;
  max-width: 1.5em;
  max-height: 1.5em;
}
