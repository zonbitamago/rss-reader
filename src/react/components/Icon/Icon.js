import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Icon.css";

class Icon extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <div className="Icon">{this.props.children}</div>;
  }
}

Icon.propTypes = {};

Icon.defaultProps = {};

export default Icon;
