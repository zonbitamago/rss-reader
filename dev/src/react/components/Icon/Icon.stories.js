import React from "react";
import { storiesOf } from "@storybook/react";
import Refresh from "@material-ui/icons/Refresh";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import IndeterminateCheckBox from "@material-ui/icons/IndeterminateCheckBox";
import Settings from "@material-ui/icons/Settings";
import Help from "@material-ui/icons/Help";

import Icon from "./Icon";

storiesOf("Atoms/Icon", module)
  .addDecorator(story => (
    <div style={{ backgroundColor: "#0D47A1", height: "3em", width: "3em" }}>
      {story()}
    </div>
  ))
  .add("Home", () => (
    <Icon>
      <Home />
    </Icon>
  ))
  .add("Refresh", () => (
    <Icon>
      <Refresh />
    </Icon>
  ))
  .add("RssFeed", () => (
    <Icon>
      <RssFeed />
    </Icon>
  ))
  .add("IndeterminateCheckBox", () => (
    <Icon>
      <IndeterminateCheckBox />
    </Icon>
  ))
  .add("Settings", () => (
    <Icon>
      <Settings />
    </Icon>
  ))
  .add("Help", () => (
    <Icon>
      <Help />
    </Icon>
  ));
