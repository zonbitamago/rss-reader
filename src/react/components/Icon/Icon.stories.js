import React from "react";
import { storiesOf } from "@storybook/react";
import Refresh from "@material-ui/icons/Refresh";
import PowerSettingsNew from "@material-ui/icons/PowerSettingsNew";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import IndeterminateCheckBox from "@material-ui/icons/IndeterminateCheckBox";
import Settings from "@material-ui/icons/Settings";
import Github from "react-icons/lib/go/mark-github";
import Twitter from "react-icons/lib/ti/social-twitter";

import Icon from "./Icon";

storiesOf("Atoms/Icon/Side", module)
  .addDecorator(story => (
    <div style={{ backgroundColor: "#01579B", height: "3em", width: "3em" }}>
      {story()}
    </div>
  ))
  .add("Power", () => (
    <Icon>
      <PowerSettingsNew />
    </Icon>
  ))
  .add("Refresh", () => (
    <Icon>
      <Refresh />
    </Icon>
  ))
  .add("RssFeed", () => (
    <Icon>
      <RssFeed />
    </Icon>
  ))
  .add("IndeterminateCheckBox", () => (
    <Icon>
      <IndeterminateCheckBox />
    </Icon>
  ))
  .add("Settings", () => (
    <Icon>
      <Settings />
    </Icon>
  ))
  .add("Help", () => (
    <Icon>
      <Github />
    </Icon>
  ));

storiesOf("Atoms/Icon/Header", module)
  .addDecorator(story => (
    <div style={{ backgroundColor: "#039BE5", height: "3em", width: "3em" }}>
      {story()}
    </div>
  ))
  .add("Home", () => (
    <Icon>
      <Home />
    </Icon>
  ))
  .add("RssFeed", () => (
    <Icon>
      <RssFeed />
    </Icon>
  ))
  .add("Twitter", () => (
    <Icon>
      <Twitter />
    </Icon>
  ));
