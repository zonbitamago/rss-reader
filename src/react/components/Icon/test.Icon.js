import React from 'react'
import { shallow } from 'enzyme'

import Icon from './Icon'

describe('Icon', () => {
  let component, props

  beforeEach(() => {
    props = {}
    component = shallow(<Icon {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})