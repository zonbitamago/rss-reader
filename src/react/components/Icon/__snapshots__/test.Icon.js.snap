// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should 1`] = `
<div
  className="Icon"
/>
`;
